app/controllers/concerns/hyrax/dashboard/collections_controller_behavior.rb

```diff
@@ -0,0 +1,279 @@
+module Hyrax
+  module Dashboard
+    module CollectionsControllerBehavior
```

Comment:
Can we make this a separate issue?  That would put a major change in that does not correspond to the original code that was being moved.  I tried to keep the structure of the code in the new location as close to the original as possible to make the PR easier to review.  It's already a bear with all the changes required for basic refactoring to a new location.